gammapy/datasets/map.py

```diff
@@ -497,9 +499,10 @@ def npred_signal(self, model_name=None):
 
         evaluators = self.evaluators
         if model_name is not None:
-            evaluators = {model_name: self.evaluators[model_name]}
+            evaluators = {name: self.evaluators[name] for name in model_name}
 
-        for evaluator in evaluators.values():
+        npred_list = []
+        for evaluator_name, evaluator in zip(evaluators.keys(), evaluators.values()):
```

Comment:
This is usually written as:
```suggestion
        for evaluator_name, evaluator in evaluators.items():
```

Reply:
done